src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveHelpers.cs

```diff
@@ -32,4 +33,16 @@ public static async Task<FileProperties> CheckFileSize(IFileStore blobDataStore,
 
         return fileProperties;
     }
+
+    public static long GetVersion(InstanceMetadata instance, bool isOriginalVersionRequested)
```

Comment:
Should this be a method in InstanceMetadata class? You can skip null check :-)

Reply:
Done